main.tf

```diff
@@ -90,24 +90,36 @@ resource "aws_s3_bucket_server_side_encryption_configuration" "default" {
 }
 
 resource "aws_s3_bucket_website_configuration" "default" {
-  for_each = local.enabled && var.website_inputs != null ? toset(var.website_inputs) : toset([])
+  for_each = local.enabled && var.website_inputs != null ? var.website_inputs : {}
```

Comment:
This cannot and should not be `for_each`, because you can only have one configuration per bucket